vignettes/terminal.Rmd

```diff
@@ -0,0 +1,100 @@
+---
+title: "Terminal Interaction"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Terminal Interaction}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(eval = FALSE)
+```
+The `rstudioapi` package provides a collection of functions that can be used to interact with the [RStudio terminal tab](https://support.rstudio.com/hc/en-us/articles/115010737148-Using-the-RStudio-Terminal).
+
+There are two primary approaches to using these functions. 
+
+1. Use `terminalExecute()` to run a specific process with the output shown in a new terminal buffer, without blocking the current R session.
+
+2. Create, query, and manipulate interactive terminals. This might be used to develop custom terminal behavior via an [RStudio addin](https://rstudio.github.io/rstudioaddins/).
+
+## TerminalExecute Scenario
+
+```{r}
+# Start a command with results displayed in a terminal buffer
+termId <- rstudioapi::terminalExecute("ping rstudio.com")
+
+# If viewing the result in the terminal buffer is sufficient,
+# then no need to do anything else. The command will continue
+# running and displaying its results without blocking the R session.
+
+# To obtain the results programmatically, wait for it to finish.
+while (is.null(rstudioapi::terminalExitCode(termId))) {
+  Sys.sleep(0.1)
+}
+
+result <- rstudioapi::terminalBuffer(termId)
+
+# Delete the buffer and close the session in the IDE
+rstudioapi::terminalKill(termId)
+```
+
+## Interative Terminal Scenario
+
+Several concepts are important to understand to make full use of these functions.
+
+### Terminal Identifier
+
+Each terminal session has a unique **terminal identifier**, a required argument for most of the functions. A terminal identifier is generated and returned when a terminal is created via `terminalCreate()` or `terminalExecute()`, and identifiers of existing terminals can be obtained via `terminalList()` or `terminalVisible()`.
+
+Note: In the current implementation, the terminal identifier is the same as the terminal caption. Thus if a terminal's caption is renamed using the RStudio user interface, the terminal identifier will change without the code knowing about it. This limitation will be fixed in a future release, and the identifier and the caption will no longer be the same; don't write code that assumes the identifier and caption are the same thing.
```

Comment:
Is this something you plan on changing for v1.1, or is that not likely to land until v1.2? If we plan to get this for v1.1 we might want to just omit this clause (or we can just make a note to ensure we update the article in v1.2 or later when this gets in)